src/metabase/models/revision/diff.clj

```diff
@@ -1,50 +1,76 @@
+
+    [:description (_ :guard some?) _]
+    (deferred-tru "changed the description")
 
     [:private true false]
-    "made it public"
+    (deferred-tru "made {0} public" identifier)
 
     [:private false true]
-    "made it private"
+    (deferred-tru "made {0} private" identifier)
+
+    [:archived false true]
+    (deferred-tru "unarchived this")
+
+    [:archived true false]
+    (deferred-tru "archived this")
+
+    [:dataset false true]
+    (deferred-tru "turned this into a model")
 
-    [:updated_at _ _]
-    nil
+    [:dataset false false]
+    (deferred-tru "changed this from a model to a saved question")
+
+    [:display _ _]
+    (deferred-tru "changed the display from {0} to {1}" v1 v2)
 
     [:result_metadata _ _]
-    nil
+    (deferred-tru "edited the metadata")
 
     [:dataset_query _ _]
-    "modified the query"
+    (deferred-tru "modified the query")
+
+    [:collection_id nil (coll-id :guard int?)]
+    (deferred-tru "moved {0} to {1}" identifier (t2/select-one-fn :name 'Collection coll-id))
+
+    [:collection_id (prev-coll-id :guard int?) (coll-id :guard int?)]
+    (deferred-tru "moved {0} from {1} to {2}"
+      identifier
+      (t2/select-one-fn :name 'Collection prev-coll-id)
+      (t2/select-one-fn :name 'Collection coll-id))
 
     [:visualization_settings _ _]
-    "changed the visualization settings"
+    (deferred-tru "changed the visualization settings")
 
-    [_ _ _]
-    (format "changed %s from \"%s\" to \"%s\"" (name k) v1 v2)))
+    :else nil))
 
 (defn build-sentence
   "Join parts of a sentence together to build a compound one."
   [parts]
   (when (seq parts)
     (cond
       (= (count parts) 1) (str (first parts) \.)
-      (= (count parts) 2) (format "%s and %s." (first parts) (second parts))
-      :else               (format "%s, %s" (first parts) (build-sentence (rest parts))))))
-
-(defn diff-string
-  "Create a string describing how `o1` is different from `o2`.
-   The directionality of the statement should indicate that `o1` changed into `o2`."
-  [t before after]
-  (when before
-    (let [ks (keys before)]
-      (some-> (filter identity (for [k ks]
-                                 (diff-string* k (k before) (k after))))
-              build-sentence
-              (str/replace-first #" it " (format " this %s " t))))))
+      (= (count parts) 2) (str (first parts) " " (deferred-tru "and")  " " (second parts) \.)
+      :else               (str (first parts) ", " (build-sentence (rest parts))))))
+
+(defn diff-strings
+  "Create a seq of string describing how `o1` is different from `o2`.
+  The directionality of the statement should indicate that `o1` changed into `o2`."
+  [model before after]
+  (let [ks (keys (or after before))]
+    (filter identity
+            (map-indexed (fn [i k]
+                           (diff-strings* k (k before) (k after)
+                                          (if (zero? i) (deferred-tru "this {0}" model) (deferred-tru "it")))) ks))))
```

Comment:
`ks` should be on a new line
```suggestion
            (map-indexed (fn [i k]
                           (diff-strings* k (k before) (k after)
                                          (if (zero? i) (deferred-tru "this {0}" model) (deferred-tru "it"))))
                         ks))))
```

Reply:
This is probably already a problem, but "this {0}" and "it" won't internationalize well. For example, "this dashboard" should translate to "la tableau de bord" and "this card" translates to "cette carte". Not sure if this is a problem worth solving right now though 🤷

Reply:
the `model` won't be translated. you can see this test for example https://github.com/metabase/metabase/blob/88fadceb53bb7bf7301b0395fc2affdc2961bf2b/test/metabase/api/revision_test.clj#L380

Reply:
Ha, I didn't notice that before. Shouldn't it be translated though?

Reply:
good question. I think it should 6430fac541